### PR TITLE
Support git stash

### LINKED ISSUE #139: Support git stash
### Command name

git

### Completion context

```shell
git stash show **<TAB>
git stash drop **<TAB>
git stash (pop|apply) **<TAB>
git stash branch <branchname> **<TAB>
git stash [push] [--] **<TAB>
git stash store **<TAB>
```

### What happens now?

Files

### What do you expect to complete?

Stashes or commit-ish where appropriate
https://git-scm.com/docs/git-stash

### PR DESCRIPTION
**Pull request by hexium310**
Closes #139.

This adds git objects for stashes into `tests/_support/git/_git`.  `git stash list` needs `tests/_support/git/_git/logs/refs/stash`.

https://git-scm.com/docs/git-stash

### CODE REVIEW

File: src/completers/git.zsh [hunk 1]
```diff
@@ -554,6 +554,69 @@ _fzf_complete_git() {
         return
     fi
 
+    if [[ $subcommand = stash ]]; then
```
Comment (chitoku-k) [inline src/completers/git.zsh:557]: Can you move this block of code after `[[ $subcommand = show ]]`? I want to maintain these conditions (loosely) alphabetically ordered.

File: src/completers/git.zsh [hunk 2]
```diff
@@ -554,6 +554,69 @@ _fzf_complete_git() {
...
+                prefix_option=$completing_option=
+            else
+                prefix_option=${prefix%%${completing_option[-1]}*}${completing_option[-1]}
+            fi
+            prefix=${prefix#$prefix_option}
+        fi
+
+        local stash_subcommand=${arguments[${arguments[(i)$subcommand]} + 1]}
+        case $stash_subcommand in
+            show)
```
Comment (hexium310) [inline src/completers/git.zsh:576]: `git stash show` can take the options that are used in `git diff`. This is parted from `apply`, `drop` and `pop` for when it is implemented.

File: src/completers/git.zsh [hunk 3]
```diff
@@ -726,7 +789,7 @@ _fzf_complete_git-commits-not-in-head() {
     local fzf_options=$1
     shift
 
-    local rev_list_all=$(git rev-list --all --oneline 2> /dev/null)
+    local rev_list_all=$(git rev-list --branches --tags --remotes --oneline 2> /dev/null)
```
Comment (hexium310) [inline src/completers/git.zsh:792]: In the case of using `--all`, `refs/stash`, which is the latest stash, is also shown.

File: src/completers/git.zsh [hunk 4]
```diff
@@ -958,6 +1021,19 @@ _fzf_complete_git-show-files_post() {
     done
 }
 
+_fzf_complete_git-stashes() {
+    local fzf_options=$1
+    shift
+
+    _fzf_complete --ansi ${(Q)${(Z+n+)fzf_options}} -- "$@$prefix_option" < <(
+        git stash list --format='%h %gd %gs' | _fzf_complete_tabularize ${fg[yellow]} $reset_color
```
Comment (chitoku-k) [inline src/completers/git.zsh:1029]: How about color the second column with green as in `git checkout **<TAB>`?

File: src/completers/git.zsh (hunk 1 above)
Comment (hexium310) [reply src/completers/git.zsh:557]: It's ok, but the order of other commands is also broken like `reset` > `commit` > `add` > `fetch`.

Comment (chitoku-k) [reply src/completers/git.zsh:557]: We might want to fix them sooner or later...

File: src/completers/git.zsh [hunk 5]
```diff
@@ -693,6 +693,69 @@ _fzf_complete_git() {
...
+                            return
+                            ;;
+
+                        --pathspec-from-file)
+                            return
+                            ;;
+                    esac
+                fi
+
+                if [[ $stash_subcommand = push ]] || [[ ${arguments[(r)--]} = -- ]]; then
```
Comment (chitoku-k) [inline src/completers/git.zsh:745]: I think `git stash **<TAB>` should be supported.

**Review by chitoku-k — APPROVED** [review]
Sweet!

**Pull request merged**